Tests/Command/ConsumeCommandTest.php

```diff
@@ -16,10 +19,30 @@ class ConsumeCommandTest extends KernelTestCase
 
     protected $mockConsumer;
 
-    public function setMockConsumer($expirationCount)
+    public function setUp()
     {
         self::bootKernel();
+    }
+
+    public function setMockQueueDriver()
+    {
+        $msg = new QueueMessage(new Message());
+
+        $this->mockQueueDriver = $this
+            ->getMockBuilder(StompQueueDriver::class)
+            ->setMethods(['connect', 'send'])
+            ->getMock();
+        $this->mockQueueDriver
+            ->method('connect');
+        $this->mockQueueDriver
+            ->method('send')
+            ->with($msg);
+
+        self::$kernel->getContainer()->set(StompQueueDriver::class, $this->mockQueueDriver);
+    }
 
+    public function setMockConsumer($expirationCount)
+    {
         $this->mockConsumer = $this
             ->getMockBuilder(QueueConsumer::class)
```

Comment:
Instead of using QueueConsumer, you could mock `ConsumerInterface` that is the interface shared for both queue consumer (sync and async)